src/test_suite/multiprocessing_utils.py

```diff
@@ -131,6 +132,43 @@ def read_fixture(fixture_file: Path) -> str | None:
     return instruction_fixture.SerializeToString(deterministic=True)
 
 
+def format_feature_set(message, indent=0, as_one_line=False):
+    lines = []
+    lines.append(" " * indent + f"FeatureSet: {{")
+    for feature in message.features:
+        lines.append(" " * (indent + 2) + hex(feature))
+    lines.append(" " * indent + "}")
+    return "\n".join(lines)
+
+
+def recursive_formatter(message, indent=0, as_one_line=False):
+    """Recursively format the message and its sub-fields"""
+    lines = []
+    for field, value in message.ListFields():
+        if field.label == field.LABEL_REPEATED:
+            lines.append(" " * indent + f"{field.name}: [")
+            for item in value:
+                if field.type == field.TYPE_MESSAGE:
+                    lines.append(recursive_formatter(item, indent + 2, as_one_line))
+                else:
+                    lines.append(" " * (indent + 2) + str(item))
+            lines.append(" " * indent + "]")
+        elif field.type == field.TYPE_MESSAGE:
+            # If the field is a message, check if it needs custom formatting
+            lines.append(" " * indent + f"{field.name}: {{")
+            if isinstance(value, FeatureSet):
+                lines.append(format_feature_set(value, indent + 2, as_one_line))
+            else:
+                # Recursively format other sub-messages
+                lines.append(recursive_formatter(value, indent + 2, as_one_line))
+            lines.append(" " * indent + "}")
+        else:
+            # Use the default formatting for non-message fields
+            lines.append(" " * indent + f"{field.name}: {value}")
+
+    return "\n".join(lines)
+
+
 def decode_single_test_case(test_file: Path) -> int:
```

Comment:
The protobuf decoded format also allowed us to execute the decoded .txt message, which was really useful for debugging (e.g. it allowed me to easily change parts of the test case by hand and see how it affected the output). Can you guard this logic behind a flag within `decode_protobuf`?